### PR TITLE
Fix flaky service instance delete e2e test

### PR DESCRIPTION
**Pull request by clintyoshimura**
## Is there a related GitHub Issue?
No

## What is this change about?
Fix service instance delete e2e test to use eventually to reduce the number of flakes

## Does this PR introduce a breaking change?

No

## Acceptance Steps

The e2e service instance delete test passes consistently

## Tag your pair, your PM, and/or team

@matt-royal

### CODE REVIEW

**Review by matt-royal — APPROVED** [review]
LGTM

File: api/tests/e2e/service_instances_test.go [hunk 1]
```diff
@@ -112,7 +112,7 @@ var _ = Describe("Service Instances", func() {
 			Expect(httpResp).To(HaveRestyStatusCode(http.StatusNotFound))
 		})
 
-		When("the user has permissions to delete service instances", func() {
+		FWhen("the user has permissions to delete service instances", func() {
```
Comment (kieron-dev) [inline api/tests/e2e/service_instances_test.go:115]: Better remove this focus!

**Review by kieron-dev — CHANGES_REQUESTED** [review]
Fine apart from the `FWhen`

**Review by tcdowney — APPROVED** [review]
LGTM

**Review by kieron-dev — APPROVED** [review]

**Pull request merged**